src/Instantiator.php

```diff
@@ -123,6 +126,16 @@ public function alwaysForceProperties(array $properties = []): self
         return $this;
     }
 
+    /**
+     * Instead of returning an instance of {@see Proxy}, use a generator to create a real proxy for the object.
+     */
+    public function withProxyGenerator(): self
```

Comment:
My original idea was using this to enable the proxy generator. However, I'm not sure if this should live in `Instantiator` or `Factory`. It's much more common in Foundry to modify `Instantiator` rather than the factory instance. On the other hand, `Factory` is currently creating the `Proxy` and the `ProxyGenerator` needs access to the factory's `Configuration` instance.

Reply:
Hmm, my first thought is `Factory` as it requires the `Configuration` as you say. Obviously given the static nature of everything it doesn't really matter but I've been trying to follow proper a dependency injection pattern where possible.